crates/lint/src/sol/med.rs

```diff
@@ -0,0 +1,68 @@
+use solar_ast::{
+    ast::{BinOp, BinOpKind, Expr, ExprKind},
+    visit::Visit,
+};
+
+use super::DivideBeforeMultiply;
+
+impl<'ast> Visit<'ast> for DivideBeforeMultiply {
+    fn visit_expr(&mut self, expr: &'ast Expr<'ast>) {
+        if let ExprKind::Binary(left_expr, BinOp { kind: BinOpKind::Mul, .. }, _) = &expr.kind {
+            if contains_division(left_expr) {
+                self.results.push(expr.span);
+            }
+        }
+
+        self.walk_expr(expr);
+    }
+}
+
+fn contains_division<'ast>(expr: &'ast Expr<'ast>) -> bool {
+    match &expr.kind {
+        ExprKind::Binary(_, BinOp { kind: BinOpKind::Div, .. }, _) => true,
+        ExprKind::Tuple(inner_exprs) => inner_exprs.iter().any(|opt_expr| {
+            if let Some(inner_expr) = opt_expr {
+                contains_division(inner_expr)
+            } else {
+                false
+            }
+        }),
+        _ => false,
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use solar_ast::{ast, visit::Visit};
+    use solar_interface::{ColorChoice, Session};
+    use std::path::Path;
+
+    use super::DivideBeforeMultiply;
+
+    #[test]
+    fn test_divide_before_multiply() -> eyre::Result<()> {
```

Comment:
If the diagnostic system is used, you can also re-use the same testing framework used in solar; this is currently not exposed but not hard to implement, publish on crates.io and re-use here

Reply:
Ive updated the PR to use the diagnostic system from `solar`. Currently, diagnostics are emitted after each lint is run within `SolidityLinter::lint()` so that all lints have consistent formatting. Do you prefer this approach or would you rather diagnostics are emitted within the individual lints instead?